packages/flutter/lib/src/widgets/form.dart

```diff
@@ -243,12 +243,42 @@ class FormState extends State<Form> {
     return _validate();
   }
 
-  bool _validate() {
+  dynamic _validate({bool isBool = true}) {
     bool hasError = false;
-    for (final FormFieldState<dynamic> field in _fields)
-      hasError = !field.validate() || hasError;
+    dynamic error;
+    for (final FormFieldState<dynamic> field in _fields) {
+      if (isBool) {
+        hasError = field.validate() != null || hasError;
+      } else {
+        error = field.validate(isBool: false);
+        if (error != null) {
+          return error;
+        }
+      }
+    }
     return !hasError;
   }
+  
+  /// Validates every [FormField] that is a descendant of this [Form], and
+  /// returns String empty if there are no errors.
+  /// ```dart
```

Comment:
nit: blank line before this

also, the example code should be wrapped in a `{@tool ...}` section.